src/main/java/mate.academy/lib/Injector.java

```
@@ -1,13 +1,84 @@
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        checkAnnotation(clazz);
+        Object implementationInstance = createNewInstance(clazz);
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                try {
+                    field.setAccessible(true);
+                    field.set(implementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Failed to initialize field value in class: "
+                            + clazz.getName()
+                            + " and field: "
+                            + field.getName(), e);
+                }
+            }
+        }
+        return implementationInstance;
+    }
+
+    private void checkAnnotation(Class<?> clazz) {
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed. "
+                    + "The @Component annotation is missing for the "
+                    + clazz.getName());
+        }
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException("Failed to create a new instance of "
+                    + clazz.getName(), e);
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        try {
+            if (interfaceClazz.isInterface()) {
+                return implementations.get(interfaceClazz);
```

Error: If the interface is not found in the `implementations` map, `implementations.get(interfaceClazz)` will return `null`. You should check for this case and throw a meaningful exception instead of returning `null`.